### PR TITLE
HZC-2052: Add SQL section to the 5.0 client sample

### PR DESCRIPTION
**Pull request by sdreger**
Output:
```
Connection Successful!
Creating a mapping...
The mapping has been created successfully.
--------------------
Deleting data via SQL...
The data has been deleted successfully.
--------------------
Inserting data via SQL...
The data has been inserted successfully.
--------------------
Retrieving all the data via SQL...
England - London
United States - Washington, DC
Czech Republic - Prague
Turkey - Ankara
Croatia - Zagreb
Australia - Canberra
--------------------
Retrieving a city name via SQL...
Country name: United States; City name: Washington, DC
--------------------
```

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -18,29 +18,93 @@ function createClientConfig() {
...
         }
+    })();
+}
+
+function sqlExample(hzClient) {
+    (async () => {
+        console.log("Creating a mapping...");
+        // See: https://docs.hazelcast.com/hazelcast/5.0/sql/mapping-to-maps
+        await hzClient.getSql().execute("DROP MAPPING cities");
+        const mappingQuery = "CREATE MAPPING cities TYPE IMap OPTIONS " +
```
Comment (ivan-shevtsov) [inline client.js]: instead of drop - create we can use "create or replace mapping"

**Review by ivan-shevtsov — APPROVED** [review]

File: client.js (hunk 1 above)
Comment (huseyinbabal) [reply client.js]: Agree with @ivan-shevtsov

Comment (sdreger) [reply client.js]: Good catch. Thanks, fixed.

File: client.js [hunk 2]
```diff
@@ -18,29 +18,92 @@ function createClientConfig() {
...
-    try {
-        const client = await Client.newHazelcastClient(createClientConfig());
-        const map = await client.getMap('map');
-        await map.put("key", "value")
-        const res = await map.get("key")
-        if (res !== "value") {
-            throw new Error("Connection failed, check your configuration.")
-        }
-        console.log("Connection Successful!");
+function mapExample(map) {
```
Comment (mdumandag) [inline client.js:21]:
```suggestion
async function mapExample(map) {
```

you could make this function async and get rid of the one nesting below, by removing the async block.

File: client.js [hunk 3]
```diff
@@ -18,29 +18,92 @@ function createClientConfig() {
...
             await map.get('key' + randomKey);
             if (++iterationCounter === 10) {
                 iterationCounter = 0;
                 map.size().then((size) => console.log(`map size: ${size}`));
             }
         }
+    })();
+}
+
+function sqlExample(hzClient) {
```
Comment (mdumandag) [inline client.js:38]: Same as above

**Review by huseyinbabal — APPROVED** [review]

File: client.js [hunk 4]
```diff
@@ -18,29 +18,92 @@ function createClientConfig() {
...
+
+(async () => {
+    try {
+        const client = await Client.newHazelcastClient(createClientConfig());
+        const map = await client.getMap('map');
+        await map.put("key", "value")
+        const res = await map.get("key")
+        if (res !== "value") {
+            throw new Error("Connection failed, check your configuration.")
+        }
```
Comment (mdumandag) [inline client.js:97]: Same as the Python client, I believe this is not necessary, as if there is a connection problem, the `await Client.newHazelcastClient(createClientConfig());` would throw

File: client.js [hunk 5]
```diff
@@ -18,29 +18,92 @@ function createClientConfig() {
...
+        console.log("Connection Successful!");
+
+        // the 'mapExample' is an example with an infinite loop inside, so if you'd like to try other examples,
+        // don't forget to comment out the following line
+        mapExample(map);
```
Comment (mdumandag) [inline client.js:102]: After making those functions async, you have to await them here

File: client_with_ssl.js [hunk 6]
```diff
@@ -30,29 +30,92 @@ function createClientConfig() {
...
-    try {
-        const client = await Client.newHazelcastClient(createClientConfig());
-        const map = await client.getMap('map');
-        await map.put("key", "value")
-        const res = await map.get("key")
-        if (res !== "value") {
-            throw new Error("Connection failed, check your configuration.")
-        }
-        console.log("Connection Successful!");
+function mapExample(map) {
```
Comment (mdumandag) [inline client_with_ssl.js:33]: Same comments as the other file

**Comment by sdreger** [comment]
@mdumandag - your PR review comments were addressed in this PR: https://github.com/hazelcast/hazelcast-cloud-nodejs-sample-client/pull/7

**Pull request merged**